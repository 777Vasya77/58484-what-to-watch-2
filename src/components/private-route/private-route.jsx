import React from 'react';
import {Route, Redirect} from 'react-router-dom';
import {selectors} from '~/selectors/selectors';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

const PrivateRoute = (props) => {
  const {component: Component, isAuth, guest} = props;
  const redirectRoute = guest ? `/` : `/login`;
  const rest = Object.assign({}, props);
  delete rest.component;
  delete rest.isAuth;

  return <Route {...rest}
    render={(routerProps) =>
      (guest ? !isAuth : isAuth)
        ? (<Component {...routerProps} />)
        : (<Redirect to={redirectRoute} />)
    }
  />;
};

PrivateRoute.propTypes = {
  component: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.object,
  ]).isRequired,
  isAuth: PropTypes.bool.isRequired,
  guest: PropTypes.bool,
};

const mapStateToProps = (state) => ({
  isAuth: selectors.user.isAuth(state)
});

export {PrivateRoute};

export default connect(mapStateToProps)(PrivateRoute);
