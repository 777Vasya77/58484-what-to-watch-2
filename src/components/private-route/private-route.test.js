import React from 'react';
import renderer from 'react-test-renderer';
import {PrivateRoute} from "~/components/private-route/private-route";

jest.mock(`react-router-dom`, () => ({
  Route: `Route`
}));

describe(`PrivateRoute component tests`, () => {
  it(`Component render correctly`, () => {
    const tree = renderer
        .create(
            <PrivateRoute component={({})} isAuth={true}/>
        ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
