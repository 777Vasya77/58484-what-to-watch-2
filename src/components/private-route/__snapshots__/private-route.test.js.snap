// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute component tests Component render correctly 1`] = `
<Route
  render={[Function]}
/>
`;
